geometry/CMakeLists.txt

```diff
@@ -37,10 +37,8 @@ set(impl_incs
 
 
 set(LIB_NAME "pcl_${SUBSYS_NAME}")
-include_directories("${CMAKE_CURRENT_SOURCE_DIR}/include")
 
 add_library(${LIB_NAME} INTERFACE)
-target_include_directories(${LIB_NAME} INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/include")
```

Comment:
Would it make sense to keep this line or replace it with `target_include_directories(${LIB_NAME} INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include> $<INSTALL_INTERFACE:${INCLUDE_INSTALL_ROOT}>)` (at least temporarily for the next release)? If I understand correctly, removing this line would _create_  the problem for geometry that https://github.com/PointCloudLibrary/pcl/pull/6006 tries to fix for kdtree and surface.

Reply:
I could also add the support for interface/header libraries in this PR?
But still keep the include_directories in PCL_SUBSYS_DEPEND.